velox/exec/Task.cpp

```diff
@@ -404,6 +404,10 @@ void Task::removeDriver(std::shared_ptr<Task> self, Driver* driver) {
         splitGroupState.clear();
         self->ensureSplitGroupsAreBeingProcessedLocked(self);
       }
+    } else {
```

Comment:
Didn't understand this change either. Please can you explain.

Reply:
We used to clear the state only in grouped execution, but ungrouped execution is no different in that regard.